### PR TITLE
Deprecate old ORA configs

### PR DESCRIPTION
**Pull request by bpoldrack**
Configuring ssh-host and base-path for an ORA remote predates ria+ URLs
and the consideration of git's URL rewrites. Those configs are also not
capable of specifying an HTTP connection let alone a separation of `url`
and `push-url`. Respect them for now if we don't find a URL, but issue a
deprecation warning. Will be removed entirely eventually.

### CODE REVIEW

File: datalad/distributed/ora_remote.py [hunk 1]
```diff
@@ -883,6 +874,39 @@ def _verify_config(self, gitdir, fail_noid=True):
...
+            self.storage_host = None if host == '0' else host
+
+            path = _get_gitcfg(gitdir,
+                               'annex.ora-remote.{}.base-path'.format(name)) or \
+                   self.annex.getconfig('base-path')
+            self.store_base_path = path.strip() if path else path
+
+            if path or host:
+                self.message("WARNING: base-path + ssh-host configs are "
+                             "deprecated and won't be considered in the future."
```
Comment (kyleam) [inline datalad/distributed/ora_remote.py:892]: Missing a space between "future." and "Use".

File: datalad/distributed/ora_remote.py [hunk 2]
```diff
@@ -883,6 +874,39 @@ def _verify_config(self, gitdir, fail_noid=True):
             self.storage_host, self.store_base_path, self.ria_store_url = \
                 verify_ria_url(self.ria_store_url, url_cfgs)
 
+        else:
+            # for now still accept the configs, if no ria-URL is known, but
+            # issue deprecation warning:
+            host = _get_gitcfg(gitdir,
+                               'annex.ora-remote.{}.ssh-host'.format(name)) or \
+                   self.annex.getconfig('ssh-host')
+            self.storage_host = None if host == '0' else host
```
Comment (kyleam) [inline datalad/distributed/ora_remote.py:883]: I think moving the deleted comment about '0' (line 916) to here could avoid some head scratching from readers, but it's obviously not very important given this code is set to be removed.

**Review by kyleam — COMMENTED** [review]
Aside from a missing space in the warning, looks good as far as I can tell.

**Review by mih — APPROVED** [review]
LGTM! Thx! Comments by @kyleam make sense.

File: datalad/distributed/ora_remote.py (hunk 1 above)
Comment (bpoldrack) [reply datalad/distributed/ora_remote.py:896]: Thx.

File: datalad/distributed/ora_remote.py (hunk 2 above)
Comment (bpoldrack) [reply datalad/distributed/ora_remote.py:887]: True. Intended to move it, but obv. forgot about it. Thx.

**Comment by bpoldrack** [comment]
Thx! Addressed the comments, only NFS stall left - hhus merging.

**Pull request merged**